File: AllSamples/Samples/Samples_Source/peer networking/peeridentity viewer/cs/dlgexportidentity.cs
//=====================================================================
// File:      DlgExportIdentity.cs
//
// Summary:   Dialog to export a PeerIdentity as a file.
//
//---------------------------------------------------------------------
// This file is part of the Microsoft .NET Framework SDK Code Samples.
// 
// Copyright (C) Microsoft Corporation.  All rights reserved.
// 
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.Net.PeerToPeer;
using System.Windows.Forms;

namespace IdentityViewer
{
    /// <summary>
    /// Dialog to export a PeerIdentity to a file.
    /// </summary>
    public class DlgExportIdentity : System.Windows.Forms.Form
    {
        private System.Windows.Forms.Label labelIdentity;
        private System.Windows.Forms.ComboBox dropListIdentity;
        private System.Windows.Forms.Label labelFile;
        private System.Windows.Forms.TextBox textBoxFile;
        private System.Windows.Forms.Button buttonBrowse;
        private System.Windows.Forms.SaveFileDialog saveFileDialog;
        private System.Windows.Forms.Label labelPassword;
        private System.Windows.Forms.TextBox textBoxPassword;
        private System.Windows.Forms.Button buttonOK;
        private System.Windows.Forms.Button buttonCancel;

        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components = null;

        /// <summary>
        ///  Constructor that creates the dialog with the PeerIdentity selected.
        /// </summary>
        /// <param name="identity">The PeerIdentity to select</param>
        public DlgExportIdentity(PeerIdentity identity)
        {
            InitializeComponent();

            Utilities.FillIdentityComboBox(dropListIdentity, identity);
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose( bool disposing )
        {
            if( disposing )
            {
                if(components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose( disposing );
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.labelIdentity = new System.Windows.Forms.Label();
            this.buttonOK = new System.Windows.Forms.Button();
            this.buttonCancel = new System.Windows.Forms.Button();
            this.textBoxFile = new System.Windows.Forms.TextBox();
            this.labelFile = new System.Windows.Forms.Label();
            this.dropListIdentity = new System.Windows.Forms.ComboBox();
            this.buttonBrowse = new System.Windows.Forms.Button();
            this.saveFileDialog = new System.Windows.Forms.SaveFileDialog();
            this.textBoxPassword = new System.Windows.Forms.TextBox();
            this.labelPassword = new System.Windows.Forms.Label();
            this.SuspendLayout();

            // 
            // labelIdentity
            // 
            this.labelIdentity.Location = new System.Drawing.Point(4, 19);
            this.labelIdentity.Name = "labelIdentity";
            this.labelIdentity.Size = new System.Drawing.Size(59, 13);
            this.labelIdentity.TabIndex = 0;
            this.labelIdentity.Text = "&Identity:";

            // 
            // buttonOK
            // 
            this.buttonOK.Anchor = System.Windows.Forms.AnchorStyles.Bottom;
            this.buttonOK.Location = new System.Drawing.Point(60, 103);
            this.buttonOK.Name = "buttonOK";
            this.buttonOK.TabIndex = 8;
            this.buttonOK.Text = "OK";
            this.buttonOK.Click += new System.EventHandler(this.buttonOK_Click);

            // 
            // buttonCancel
            // 
            this.buttonCancel.Anchor = System.Windows.Forms.AnchorStyles.Bottom;
            this.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.buttonCancel.Location = new System.Drawing.Point(156, 103);
            this.buttonCancel.Name = "buttonCancel";
            this.buttonCancel.TabIndex = 9;
            this.buttonCancel.Text = "Cancel";
            this.buttonCancel.Click += new System.EventHandler(this.buttonCancel_Click);

            // 
            // textBoxFile
            // 
            this.textBoxFile.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) | System.Windows.Forms.AnchorStyles.Right)));
            this.textBoxFile.Location = new System.Drawing.Point(61, 47);
            this.textBoxFile.Name = "textBoxFile";
            this.textBoxFile.Size = new System.Drawing.Size(196, 19);
            this.textBoxFile.TabIndex = 4;

            // 
            // labelFile
            // 
            this.labelFile.Location = new System.Drawing.Point(4, 47);
            this.labelFile.Name = "labelFile";
            this.labelFile.Size = new System.Drawing.Size(59, 13);
            this.labelFile.TabIndex = 3;
            this.labelFile.Text = "&File:";

            // 
            // dropListIdentity
            // 
            this.dropListIdentity.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) | System.Windows.Forms.AnchorStyles.Right)));
            this.dropListIdentity.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.dropListIdentity.Location = new System.Drawing.Point(61, 19);
            this.dropListIdentity.Name = "dropListIdentity";
            this.dropListIdentity.Size = new System.Drawing.Size(226, 21);
            this.dropListIdentity.Sorted = true;
            this.dropListIdentity.TabIndex = 1;

            // 
            // buttonBrowse
            // 
            this.buttonBrowse.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Right)));
            this.buttonBrowse.Location = new System.Drawing.Point(265, 47);
            this.buttonBrowse.Name = "buttonBrowse";
            this.buttonBrowse.Size = new System.Drawing.Size(22, 19);
            this.buttonBrowse.TabIndex = 5;
            this.buttonBrowse.Text = "...";
            this.buttonBrowse.Click += new System.EventHandler(this.buttonBrowse_Click);

            // 
            // saveFileDialog
            // 
            this.saveFileDialog.DefaultExt = "idx";
            this.saveFileDialog.Filter = "Exported identity files|*.idx|All files|*.*";
            this.saveFileDialog.Title = "Save Identity Information as...";

            // 
            // textBoxPassword
            // 
            this.textBoxPassword.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) | System.Windows.Forms.AnchorStyles.Right)));
            this.textBoxPassword.Location = new System.Drawing.Point(61, 73);
            this.textBoxPassword.Name = "textBoxPassword";
            this.textBoxPassword.PasswordChar = '*';
            this.textBoxPassword.Size = new System.Drawing.Size(226, 19);
            this.textBoxPassword.TabIndex = 7;
            this.textBoxPassword.UseSystemPasswordChar = true;

            // 
            // labelPassword
            // 
            this.labelPassword.Location = new System.Drawing.Point(4, 73);
            this.labelPassword.Name = "labelPassword";
            this.labelPassword.Size = new System.Drawing.Size(59, 13);
            this.labelPassword.TabIndex = 6;
            this.labelPassword.Text = "&Password:";

            // 
            // DlgExportIdentity
            // 
            this.AcceptButton = this.buttonOK;
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.CancelButton = this.buttonCancel;
            this.ClientSize = new System.Drawing.Size(290, 137);
            this.Controls.Add(this.textBoxPassword);
            this.Controls.Add(this.labelPassword);
            this.Controls.Add(this.buttonBrowse);
            this.Controls.Add(this.dropListIdentity);
            this.Controls.Add(this.textBoxFile);
            this.Controls.Add(this.labelFile);
            this.Controls.Add(this.buttonCancel);
            this.Controls.Add(this.buttonOK);
            this.Controls.Add(this.labelIdentity);
            this.MaximizeBox = false;
            this.MaximumSize = new System.Drawing.Size(999, 164);
            this.MinimizeBox = false;
            this.MinimumSize = new System.Drawing.Size(298, 164);
            this.Name = "DlgExportIdentity";
            this.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide;
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Export Identity";
            this.ResumeLayout(false);
            this.PerformLayout();
        }
        #endregion

        /// <summary>
        ///   Try to export the selected PeerIdentity to the file.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void buttonOK_Click(object sender, System.EventArgs e)
        {
            if (ExportIdentity())
            {
                Close();
            }
        }

        /// <summary>
        /// Try to export the identity.
        /// </summary>
        /// <returns>True if the identity could be exported.</returns>
        private bool ExportIdentity()
        {
            bool success = false;

            try
            {
                PeerIdentity identity = (PeerIdentity)dropListIdentity.SelectedItem;
                string exportedData = identity.Export(textBoxPassword.Text);
                Utilities.WriteTextFile(textBoxFile.Text, exportedData);
                success = true;
            }
            catch (Exception e)
            {
                Utilities.DisplayException(e, this);
            }

            return success;
        }

        /// <summary>
        /// Close the dialog without exporting any information.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void buttonCancel_Click(object sender, System.EventArgs e)
        {
            Close();
        }
        
        /// <summary>
        ///  Select a filename.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void buttonBrowse_Click(object sender, System.EventArgs e)
        {
            if (textBoxFile.TextLength == 0)
            {
                // default the file name to the selected identity.
                saveFileDialog.FileName = dropListIdentity.Text;
            }

            if (DialogResult.OK == saveFileDialog.ShowDialog(this))
            {
                textBoxFile.Text = saveFileDialog.FileName;
            }
        }
    }
}

File: AllSamples/Samples/Samples_Source/peer networking/peeridentity viewer/vb/dlgexportidentity.vb
'=====================================================================
'  File:      DlgExportIdentity.vb
'
'  Summary:   Dialog to export a PeerIdentity as a file.
'
'---------------------------------------------------------------------
' This file is part of the Microsoft .NET Framework SDK Code Samples.
' 
' Copyright (C) Microsoft Corporation.  All rights reserved.
' 
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Option Explicit On
Option Strict On

Imports System
Imports System.ComponentModel
Imports System.Net.PeerToPeer
Imports System.Windows.Forms

Public Class DlgExportIdentity
    Inherits System.Windows.Forms.Form

    Private dropListIdentity As System.Windows.Forms.ComboBox
    Private textBoxFile As System.Windows.Forms.TextBox
    Private textBoxPassword As System.Windows.Forms.TextBox
    Private saveFileDialog As System.Windows.Forms.SaveFileDialog
    Private WithEvents buttonOK As System.Windows.Forms.Button
    Private WithEvents buttonCancel As System.Windows.Forms.Button
    Private WithEvents buttonBrowse As System.Windows.Forms.Button
    Private WithEvents labelIdentity As System.Windows.Forms.Label
    Private WithEvents labelFile As System.Windows.Forms.Label
    Private WithEvents labelPassword As System.Windows.Forms.Label

    Private components As System.ComponentModel.Container = Nothing

    '  Constructor that creates the dialog with the PeerIdentity selected.
    '
    Public Sub New(ByVal identity As PeerIdentity)
        InitializeComponent()

        Utilities.FillIdentityComboBox(dropListIdentity, identity)

    End Sub 'New


    ' Clean up any resources being used.
    '
    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)

    End Sub 'Dispose

#Region "Windows Form Designer generated code"

    ' <summary>
    ' Required method for Designer support - do not modify
    ' the contents of this method with the code editor.
    ' </summary>
    Private Sub InitializeComponent()
        Me.labelIdentity = New System.Windows.Forms.Label
        Me.buttonOK = New System.Windows.Forms.Button
        Me.buttonCancel = New System.Windows.Forms.Button
        Me.textBoxFile = New System.Windows.Forms.TextBox
        Me.labelFile = New System.Windows.Forms.Label
        Me.dropListIdentity = New System.Windows.Forms.ComboBox
        Me.buttonBrowse = New System.Windows.Forms.Button
        Me.saveFileDialog = New System.Windows.Forms.SaveFileDialog
        Me.textBoxPassword = New System.Windows.Forms.TextBox
        Me.labelPassword = New System.Windows.Forms.Label
        Me.SuspendLayout()
        '
        'labelIdentity
        '
        Me.labelIdentity.Location = New System.Drawing.Point(4, 19)
        Me.labelIdentity.Name = "labelIdentity"
        Me.labelIdentity.Size = New System.Drawing.Size(59, 13)
        Me.labelIdentity.TabIndex = 0
        Me.labelIdentity.Text = "&Identity:"
        '
        'buttonOK
        '
        Me.buttonOK.Anchor = System.Windows.Forms.AnchorStyles.Bottom
        Me.buttonOK.Location = New System.Drawing.Point(60, 103)
        Me.buttonOK.Name = "buttonOK"
        Me.buttonOK.TabIndex = 8
        Me.buttonOK.Text = "OK"
        '
        'buttonCancel
        '
        Me.buttonCancel.Anchor = System.Windows.Forms.AnchorStyles.Bottom
        Me.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.buttonCancel.Location = New System.Drawing.Point(156, 103)
        Me.buttonCancel.Name = "buttonCancel"
        Me.buttonCancel.TabIndex = 9
        Me.buttonCancel.Text = "Cancel"
        '
        'textBoxFile
        '
        Me.textBoxFile.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.textBoxFile.Location = New System.Drawing.Point(61, 47)
        Me.textBoxFile.Name = "textBoxFile"
        Me.textBoxFile.Size = New System.Drawing.Size(196, 19)
        Me.textBoxFile.TabIndex = 4
        '
        'labelFile
        '
        Me.labelFile.Location = New System.Drawing.Point(4, 47)
        Me.labelFile.Name = "labelFile"
        Me.labelFile.Size = New System.Drawing.Size(59, 13)
        Me.labelFile.TabIndex = 3
        Me.labelFile.Text = "&File:"
        '
        'dropListIdentity
        '
        Me.dropListIdentity.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.dropListIdentity.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
        Me.dropListIdentity.Location = New System.Drawing.Point(61, 19)
        Me.dropListIdentity.Name = "dropListIdentity"
        Me.dropListIdentity.Size = New System.Drawing.Size(226, 21)
        Me.dropListIdentity.Sorted = True
        Me.dropListIdentity.TabIndex = 1
        '
        'buttonBrowse
        '
        Me.buttonBrowse.Anchor = CType((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.buttonBrowse.Location = New System.Drawing.Point(265, 47)
        Me.buttonBrowse.Name = "buttonBrowse"
        Me.buttonBrowse.Size = New System.Drawing.Size(22, 19)
        Me.buttonBrowse.TabIndex = 5
        Me.buttonBrowse.Text = "..."
        '
        'saveFileDialog
        '
        Me.saveFileDialog.DefaultExt = "idx"
        Me.saveFileDialog.Filter = "Exported identity files|*.idx|All files|*.*"
        Me.saveFileDialog.Title = "Save Identity Information as..."
        '
        'textBoxPassword
        '
        Me.textBoxPassword.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.textBoxPassword.Location = New System.Drawing.Point(61, 73)
        Me.textBoxPassword.Name = "textBoxPassword"
        Me.textBoxPassword.Size = New System.Drawing.Size(226, 19)
        Me.textBoxPassword.TabIndex = 7
        '
        'labelPassword
        '
        Me.labelPassword.Location = New System.Drawing.Point(4, 73)
        Me.labelPassword.Name = "labelPassword"
        Me.labelPassword.Size = New System.Drawing.Size(59, 13)
        Me.labelPassword.TabIndex = 6
        Me.labelPassword.Text = "&Password:"
        '
        'DlgExportIdentity
        '
        Me.AcceptButton = Me.buttonOK
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.CancelButton = Me.buttonCancel
        Me.ClientSize = New System.Drawing.Size(290, 137)
        Me.Controls.Add(Me.textBoxPassword)
        Me.Controls.Add(Me.labelPassword)
        Me.Controls.Add(Me.buttonBrowse)
        Me.Controls.Add(Me.dropListIdentity)
        Me.Controls.Add(Me.textBoxFile)
        Me.Controls.Add(Me.labelFile)
        Me.Controls.Add(Me.buttonCancel)
        Me.Controls.Add(Me.buttonOK)
        Me.Controls.Add(Me.labelIdentity)
        Me.MaximizeBox = False
        Me.MaximumSize = New System.Drawing.Size(999, 164)
        Me.MinimizeBox = False
        Me.MinimumSize = New System.Drawing.Size(298, 164)
        Me.Name = "DlgExportIdentity"
        Me.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
        Me.Text = "Export Identity"
        Me.ResumeLayout(False)
        Me.PerformLayout()

    End Sub 'InitializeComponent
#End Region

    '  Try to export the selected PeerIdentity to the file.
    '
    Private Sub buttonOK_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonOK.Click
        If ExportIdentity() Then
            Me.Close()
        End If

    End Sub 'buttonOK_Click

    ' Try to export the identity.
    '
    Private Function ExportIdentity() As Boolean
        Dim success As Boolean = False

        Try
            Dim identity As PeerIdentity = CType(dropListIdentity.SelectedItem, PeerIdentity)
            Dim exportedData As String = identity.Export(textBoxPassword.Text)
            Utilities.WriteTextFile(textBoxFile.Text, exportedData)
            success = True
        Catch e As Exception
            Utilities.DisplayException(e, Me)
        End Try

        Return success

    End Function 'ExportIdentity

    ' Close the dialog without exporting any information.
    '
    Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonCancel.Click
        Me.Close()
    End Sub 'buttonCancel_Click

    '  Select a filename.
    '
    Private Sub buttonBrowse_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonBrowse.Click
        If textBoxFile.TextLength = 0 Then
            ' default the file name to the selected identity.
            saveFileDialog.FileName = dropListIdentity.Text
        End If

        If DialogResult.OK = saveFileDialog.ShowDialog(Me) Then
            textBoxFile.Text = saveFileDialog.FileName
        End If

    End Sub 'buttonBrowse_Click

End Class 'DlgExportIdentity